shared/src/sources/balancer/event_handler.rs

```diff
@@ -44,6 +48,8 @@ pub struct BalancerPoolRegistry {
         Mutex<EventHandler<Web3, BalancerV2WeightedPoolFactoryContract, PoolStorage>>,
     two_token_pool_updater:
         Mutex<EventHandler<Web3, BalancerV2WeightedPool2TokensFactoryContract, PoolStorage>>,
+    stable_pool_updater:
+        Mutex<EventHandler<Web3, BalancerV2StablePoolFactoryContract, PoolStorage>>,
```

Comment:
Maybe it makes sense to have separate registries for weighted pools and stable pools? Not suggesting doing it, just wondering if you considered it and what the trade-offs were.

The reason I'm suggesting this is that I see a bunch of `{Weighted,Stable}Pool -> RegisteredPool -> {Weighted,Stable}Pool` conversions we might be able to avoid with separate registries.